ast/ast.go

```diff
@@ -532,6 +535,56 @@ func (*StringLiteral) expression()          {}
 func (*UnaryExpression) expression()        {}
 func (*UnsignedIntegerLiteral) expression() {}
 
+type StringExpression struct {
+	BaseNode
+
+	Parts []*StringExpressionPart `json:"parts"`
+}
+
+func (*StringExpression) Type() string { return "StringExpression" }
+
+func (e *StringExpression) Copy() Node {
+	if e == nil {
+		return e
+	}
+	ne := new(StringExpression)
+	*ne = *e
+	ne.BaseNode = e.BaseNode.Copy()
+
+	if len(e.Parts) > 0 {
+		ne.Parts = make([]*StringExpressionPart, len(e.Parts))
+		for i, p := range e.Parts {
+			ne.Parts[i] = p.Copy().(*StringExpressionPart)
+		}
+	}
+
+	return ne
+}
+
+type StringExpressionPart struct {
```

Comment:
Should `StringExpressionPart` be an interface instead of a struct? It seems to behave like a union type.

Reply:
I think a struct representing a string part makes more sense vs separate `Text` and `${ ... }` ast nodes. In my opinion.

Reply:
To keep parity with the Rust implementation where `StringExpressionPart` is implemented as an `enum` type, I did indeed have to make it an interface type in the Go implementation.